### ISSUE TITLE
ExecuteSqlCommand keeps using IsolationLevel of previous transaction 

### ISSUE THREAD

**Issue by Kloizdena** [dotnet/efcore#16902]
### Steps to reproduce
Start the following program (tested with .NET FW 4.7.2 & .NET Core 2.2):

```c#
using Microsoft.EntityFrameworkCore;
using System;
using System.Threading;

namespace IsolationLevelTest
{
    class Program
    {
        static void Main(string[] args)
        {
            var opts = new DbContextOptionsBuilder<MyContext>().UseSqlServer("Data Source=.;Initial Catalog=Test;Integrated Security=True", o => o.CommandTimeout(900)).Options;

            using (var context = new MyContext(opts))
            using (var transaction = context.Database.BeginTransaction(System.Data.IsolationLevel.Serializable))
            {
                context.Database.ExecuteSqlCommand("SELECT 1");
                transaction.Commit();
            }
            var cts = new CancellationTokenSource();
            for (int i = 0; i < 10; i++)
            {
                var t = new Thread(new ThreadStart(() =>
                {
                    while (!cts.IsCancellationRequested)
                        using (var context = new MyContext(opts))
                        {
                            context.Database.ExecuteSqlCommand("WAITFOR DELAY '00:00:01'");
                        }
                }));
                t.Start();
            }
            Console.WriteLine("Started!");
            Console.ReadLine();
            cts.Cancel();
        }
    }

    class MyContext : DbContext
    {
        public MyContext(DbContextOptions options) : base(options)
        {
        }
    }
}
```

While the program is running, query the isolation levels of the active sessions in SQL:
```sql
select s.transaction_isolation_level, t.*, *
from sys.dm_exec_sessions s
join sys.dm_exec_requests r on s.session_id = r.session_id
cross apply sys.dm_exec_sql_text(r.sql_handle) t
where host_name is not null
and s.session_id != @@SPID
and program_name like '%.Net SqlClient Data Provider'
```
Expected: all the sessions executing the `WAITFOR DELAY` command should use isolation level 2 (ReadCommitted).  
Actual: one of the sessions uses isolation level 4 (Serializable). The other sessions use level 2 (ReadCommitted).
![image](https://user-images.githubusercontent.com/17968888/62334559-6dcb2700-b4c8-11e9-8785-be6d416c9f33.png)

| transaction_isolation_level | dbid | objectid | number | encrypted | text                     | session_id | login_time              | program_name                 | host_process_id | client_version | client_interface_name        |
|-----------------------------|------|----------|--------|-----------|--------------------------|------------|-------------------------|------------------------------|-----------------|----------------|------------------------------|
| 4                           | 8    | NULL     | NULL   | 0         | WAITFOR DELAY '00:00:01' | 52         | 2019-08-02 01:41:44.937 | .Net SqlClient Data Provider | 7708            | 7              | .Net SqlClient Data Provider |
| 2                           | 8    | NULL     | NULL   | 0         | WAITFOR DELAY '00:00:01' | 54         | 2019-08-02 01:41:44.997 | .Net SqlClient Data Provider | 7708            | 7              | .Net SqlClient Data Provider |
| 2                           | 8    | NULL     | NULL   | 0         | WAITFOR DELAY '00:00:01' | 56         | 2019-08-02 01:41:45.060 | .Net SqlClient Data Provider | 7708            | 7              | .Net SqlClient Data Provider |
| 2                           | 8    | NULL     | NULL   | 0         | WAITFOR DELAY '00:00:01' | 57         | 2019-08-02 01:41:45.110 | .Net SqlClient Data Provider | 7708            | 7              | .Net SqlClient Data Provider |
| 2                           | 8    | NULL     | NULL   | 0         | WAITFOR DELAY '00:00:01' | 58         | 2019-08-02 01:41:45.137 | .Net SqlClient Data Provider | 7708            | 7              | .Net SqlClient Data Provider |
| 2                           | 8    | NULL     | NULL   | 0         | WAITFOR DELAY '00:00:01' | 59         | 2019-08-02 01:41:45.160 | .Net SqlClient Data Provider | 7708            | 7              | .Net SqlClient Data Provider |
| 2                           | 8    | NULL     | NULL   | 0         | WAITFOR DELAY '00:00:01' | 60         | 2019-08-02 01:41:45.170 | .Net SqlClient Data Provider | 7708            | 7              | .Net SqlClient Data Provider |
| 2                           | 8    | NULL     | NULL   | 0         | WAITFOR DELAY '00:00:01' | 61         | 2019-08-02 01:41:45.180 | .Net SqlClient Data Provider | 7708            | 7              | .Net SqlClient Data Provider |
| 2                           | 8    | NULL     | NULL   | 0         | WAITFOR DELAY '00:00:01' | 62         | 2019-08-02 01:41:45.187 | .Net SqlClient Data Provider | 7708            | 7              | .Net SqlClient Data Provider |
| 2                           | 8    | NULL     | NULL   | 0         | WAITFOR DELAY '00:00:01' | 63         | 2019-08-02 01:41:45.210 | .Net SqlClient Data Provider | 7708            | 7              | .Net SqlClient Data Provider |

Workaround: surround ExecuteSqlCommand with a transaction created with the correct isolation level:
```c#
using (var transaction = context.Database.BeginTransaction(System.Data.IsolationLevel.ReadCommitted))
{
    context.Database.ExecuteSqlCommand("WAITFOR DELAY '00:00:01'");
    transaction.Commit();
}
```

### Further technical details
EF Core version: 2.2.6
Database Provider: Microsoft.EntityFrameworkCore.SqlServer
Operating system: Windows 10.0.18362.239
IDE: Visual Studio 2019 16.2.0

**Comment by AndriySvyryd** [comment]
@roji Do you want to take a look?

**Comment by roji** [comment]
Yep, will do.

**Comment by roji** [comment]
This is an issue with SqlClient and not related to Entity Framework in any way, see https://github.com/dotnet/SqlClient/issues/96.

tl;dr when SqlConnections are returned to the pool, their isolation level isn't reset. The next time the connection is handed out, it has whatever isolation level it last had in use. Until this issue is fixed, the workaround is as you've done - never assume that a newly-opened connection has any particular isolation level, and always set it explicitly.
